eo-maven-plugin/src/test/java/org/eolang/maven/objectionary/OyFallbackTest.java

```diff
@@ -23,72 +23,148 @@
         );
-        final String object = "org.example.main";
         MatcherAssert.assertThat(
-            objectionary.contains(object),
-            Matchers.is(false)
+            this.primary.invocations.get(),
+            Matchers.equalTo(1)
         );
         MatcherAssert.assertThat(
-            new TextOf(objectionary.get(object)).asString(),
-            Matchers.is(Matchers.notNullValue())
+            this.secondary.invocations.get(),
+            Matchers.equalTo(0)
         );
+    }
+
+    @Test
+    void getsObjectWhenPrimaryNotContains() throws Exception {
         MatcherAssert.assertThat(
-            path.resolve("pulled/master/org/example/main.eo").toFile().exists(),
-            Matchers.is(true)
+            new TextOf(this.fallback.get("org.example.secondary")).asString(),
+            Matchers.equalTo(OyFallbackTest.SOURCE)
+        );
+        MatcherAssert.assertThat(
+            this.primary.invocations.get(),
+            Matchers.equalTo(1)
+        );
+        MatcherAssert.assertThat(
+            this.secondary.invocations.get(),
+            Matchers.equalTo(1)
         );
+    }
+
+    @Test
+    void containsObject() throws IOException {
         MatcherAssert.assertThat(
-            objectionary.contains(object),
+            this.fallback.contains("org.example.main"),
             Matchers.is(true)
         );
         MatcherAssert.assertThat(
-            objectionary.get(object),
-            Matchers.is(Matchers.notNullValue())
+            this.fallback.contains("org.example.absent"),
+            Matchers.is(false)
+        );
+        MatcherAssert.assertThat(
+            this.primary.invocations.get(),
+            Matchers.equalTo(2)
         );
         MatcherAssert.assertThat(
-            counter.get(),
-            Matchers.is(2)
+            this.secondary.invocations.get(),
+            Matchers.equalTo(1)
         );
     }
+
+    @Test
+    void convertsToString() {
+        MatcherAssert.assertThat(
+            this.fallback.toString(),
+            Matchers.equalTo(String.format("[%s]+[fallback to %s]", this.primary, this.secondary))
+        );
+    }
+
+    /**
+     * Mock objectionary.
+     *
+     * @since 0.29.0
+     */
+    private static final class OyMock implements Objectionary {
+
+        /**
+         * Invocations counter.
+         */
+        private final AtomicInteger invocations;
+
+        /**
+         * Object name.
+         */
+        private final String name;
+
+        /**
+         * Ctor.
+         * @param object Object name
+         */
+        private OyMock(final String object) {
+            this.name = object;
+            this.invocations = new AtomicInteger();
+        }
+
+        @Override
+        public Input get(final String object) throws IOException {
+            this.invocations.incrementAndGet();
+            if (this.name.equals(object)) {
+                return new InputOf(OyFallbackTest.SOURCE);
```

Comment:
@volodya-lombrozo I believe it's better to make the `source` internal to `OyMock`. Let's pass the source as another constructor param and eliminate dependency on `OyFallbackTest`

Reply:
@mximp Indeed, It makes sense. I've changed the code a little bit. Can you take a look again, please?